pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -32,12 +32,19 @@ type clients struct {
 	computeClient *gophercloud.ServiceClient
 }
 
+// Flavor embeds information from the Gophercloud Flavor struct, plus the logic
+// needed for determining whether a flavor is of baremetal type.
+type Flavor struct {
+	*flavors.Flavor
+	GetExtraSpec func(key string) (value string, ok bool, err error)
```

Comment:
why embed a function type instead of defining a member function?

Also it looks like you are using closure here https://github.com/openshift/installer/pull/4222/files#diff-d118447e547aad067bf40359daebf932R160-R167 to keep access to the compute client? tieing the lifecycle in some complex/not-easily-traceble ways. 

My personal preference would be just define the requirements in the struct and define a function on the struct.

_just a recommendation_

Reply:
Your recommendation pushes me to simplify the code; it's most often a good thing. Thanks!

I have traded lazy-loading for grater specialisation; let's leave generalisation for when we need it.